lib/Doctrine/ORM/Mapping/ClassMetadataInfo.php

```diff
@@ -194,6 +194,12 @@ class ClassMetadataInfo implements ClassMetadata
      */
     public const FETCH_EXTRA_LAZY = 4;
 
+    /**
+     * Specifies that all entities of a collection valued association are fetched using a single or more
+     * additional WHERE IN queries.
+     */
+    public const FETCH_SUBSELECT = 5;
```

Comment:
I'm very late to this party, but I think I like `FETCH_EAGER_BATCHED` better. There seems to be 2 families of fetch mode, eager and lazy, indicating which one we are dealing with is important. The `BATCHED` part indicates that we are slicing it, which sounds like an implementation detail though… so maybe `FETCH_EAGER_SUBSELECT`? The fact that entities are fetched afterwards seems more important than the fact that it is sliced in batches, and specifying that this is done with a `SUBSELECT` a.k.a.`WHERE IN` will lead to understanding that. Or we could be explicit and name it something like `FETCH_EAGER_AFTERWARDS`

Reply:
Forgot to remove this, as the feature is now just a change of the FETCH_EAGER behavior for 1:n assoc.